### ISSUE TITLE
unicorn/no-array-push-push - breaks stream.push

### ISSUE THREAD

**Issue by dylang** [sindresorhus/eslint-plugin-unicorn#1057]
`unicorn/no-array-push-push` combines `.push`, but it can't do this if the `.push` is for something that is not an array, like `stream`.

## Before
```ts
const stream = new Readable();
stream.push('attachment content');
stream.push(null); // end of file
```

## After
```ts
const stream = new Readable();
stream.push('attachment content', null); // end of file
//                                ^^^^ null is not a valid encoding
```

**Comment by fisker** [comment]
I'm sorry for this, we can't know the type of the object, in this case we can check if the variable is constructed by other class, but not very meaningful since the object may from other file or function, I guess you'll have to disable this rule for this.

**Comment by sindresorhus** [comment]
@fisker I think we could detect this case. It's quite unusual to push `null`, so if the last has `null`, we could just not report.

**Comment by fisker** [comment]
But there will still

```js
const stream = new Readable();
stream.push('one string');
stream.push('another string');  // <- this will be reported too
stream.push(null); // end of file
```

**Comment by sindresorhus** [comment]
I meant that if the last `.push` has `null`, we don't report any of the `.push` calls on the instance.

**Comment by fisker** [comment]
I understand that, my point is doing this is not really very useful, 

```js
const stream = new Readable();
stream.push('one string');
stream.push('another string');  // <- this will be reported too
```

**Comment by sindresorhus** [comment]
But you have to end the stream with `null`. Alternatively, we could just ignore if the instance is called `stream`.
